### PR TITLE
Remove Java files from default Maven Groovy formatting

### LINKED ISSUE #902: Remove *.java files from default Maven Groovy formatting
This is the default `includes` of Maven Groovy:
```xml
<includes>
  <include>src/main/groovy/**/*.groovy</include>
  <include>src/test/groovy/**/*.groovy</include>
  <include>src/main/java/**/*.java</include>
  <include>src/test/java/**/*.java</include>
</includes>
```

I'm not sure why Maven Groovy is configured to format .java files. At first, it stripped all the semicolons from my .java files. Next, I couldn't figure out why `mvn spotless:apply` wasn't working, when it turned out that Groovy formatting kept overwriting the .java files immediately after Java formatting.

### PR DESCRIPTION
**Pull request by lutovich**
All Groovy formatters will default to formatting only the following files:
 * `src/main/groovy/**/*.groovy`
 * `src/test/groovy/**/*.groovy`

They will not touch any `.*java` files.

Fixes #902.

### CODE REVIEW

**Comment by lutovich** [comment]
It feels like Groovy formatters should not touch Java files by default. However, I do not know much about the groovy-eclipse formatter. This might be a backward-incompatible change. Any input is much appreciated!

**Comment by lutovich** [comment]
Hi @bitwiseman, could you please take a look at this PR and the associated issue? Thanks!

**Review by bitwiseman — APPROVED** [review]

File: plugin-maven/CHANGES.md [hunk 1]
```diff
@@ -4,7 +4,14 @@ We adhere to the [keepachangelog](https://keepachangelog.com/en/1.0.0/) format (
 
 ## [Unreleased]
 ### Fixed
-* Remove Java files from default Maven Groovy formatting ([#1051](https://github.com/diffplug/spotless/pull/1051)).
+* **BREAKING** Remove Java files from default Maven Groovy formatting ([#1051](https://github.com/diffplug/spotless/pull/1051)).
+  * This breaking change affects you only if you are using the `<groovy>` language block **and** you have `.java` files located within `src/main/groovy` or `src/test/groovy`.
+  * Before `3.0.0`, the default target of groovy was
+    * `src/main/groovy/**/*.groovy`
+    * `src/test/groovy/**/*.groovy`
+    * `src/main/groovy/**/*.java`
```
Comment (lutovich) [inline plugin-maven/CHANGES.md:12]: Just a small correction. I removed the following file masks:
 * `src/main/java/**/*.java`
 * `src/test/java/**/*.java`

The Groovy formatter tried to format Java sources that live under the Java sources root. I think it is weird and perhaps even undesirable to format such source files with a Groovy formatter. This PR changes the default behavior, but it is more like a fix in my opinion.

Comment (nedtwigg) [reply plugin-maven/CHANGES.md:12]: Oh man that is super weird! It's so weird my brain couldn't read it the first time, thanks for pointing that out! I always agreed that it was a fix, but now I also agree that it is not a "breaking fix".

**Comment by lutovich** [comment]
Thank you @nedtwigg! I'll merge this PR.

**Comment by lutovich** [comment]
`main` branch build is failing after the merge. I will investigate.

**Comment by nedtwigg** [comment]
Released in `2.19.0`.

**Pull request merged**